modules/nlu/src/backend/training-pipeline.ts

```diff
@@ -471,7 +482,8 @@ const TrainOutOfScope = async (input: TrainStep, tools: Tools, progress: progres
 
   const oos_points = featurizeOOSUtterances(noneUtts, input.vocabVectors, tools)
   let combinedProgress = 0
-  const ctxModels: [string, string][] = await Promise.map(input.ctxToTrain, async ctx => {
+  const ctxModels: [string, string][] = []
+  for (const ctx of input.ctxToTrain) {
```

Comment:
Promise.map was parallel now it's sequential

Reply:
:+1: